### ISSUE TITLE
flow vs typescript

### ISSUE THREAD

**Issue by anithri** [jondot/hygen#150]
As I gear up for a bunch of programming, I need to get my head around flow.  And before I do that, I want to make sure that typescript is not a better choice.

I have near zero experience with either, so I have to learn something anyways.  I hear bitching about config and performance for flow in WebStorm, and i have yet to be able to have a linted error free pane in my editor because something is always not right for flow.

Thoughts?

**Comment by jondot** [comment]
Hi @anithri I have a goal to rewrite/convert Hygen to Typescript (that's what I'm prepping for). Flow isn't such a long-term bet anymore, unfortunately.

**Comment by anithri** [comment]
Played with the last night.  It's easier than I thought, I like what I've seen so far.  

I quit with a couple of hundred errors still showing up via tsc, and it was in no way shape or form an organized effort.  I think the dominating issue is the web of types that have to be created.

Also, completely ripping out flow and what is there of typescript, and then making sure the tests run.  Then reinstall it from scratch so you remember how it's done.

I found having to rewrite the web of types frustrating, But I've enjoyed starting fresh on that idea i'm fleshing out.  

We could use an architectural plan of the types.  where are the namespace set, how a file is laid out, what syntax to use. where the files are located. 
```typescript
// src/hygen/index.d.ts
export * from '../utils' /* src/utils/index.d.ts */
export * from '../resolvers'  /* src/resolvers/index.d.ts */
---
// src/utils/index.d.ts
export declare class Logger {
}
```
